python/ray/air/result.py

```diff
@@ -21,12 +23,23 @@ class Result:
     Args:
         metrics: The final metrics as reported by an Trainable.
         checkpoint: The final checkpoint of the Trainable.
+        best_checkpoint: The best checkpoint of the Trainable.
```

Comment:
This behavior is a bit confusing. Do we need to expose all these details to the user?